apps/arena/native/physics/src/collision_detection/ear_clipping.rs

```diff
@@ -0,0 +1,195 @@
+
+    get_cross_product_value(&current_to_previous_vector, &current_to_next_vector) > 0.0
+        && !triangle_has_point_inside
+}
+
+// Check if any vertex beside the one that belongs to the triangle is inside the triangle area
+fn triangle_has_point_inside(triangle: &Entity, vertex: &Vec<Position>) -> bool {
+    let vector_from_a_to_b = Position::sub(&triangle.vertices[1], &triangle.vertices[0]);
+    let vector_from_b_to_c = Position::sub(&triangle.vertices[2], &triangle.vertices[1]);
+    let vector_from_c_to_a = Position::sub(&triangle.vertices[0], &triangle.vertices[2]);
+
+    let mut result = false;
+    for v in vertex {
+        if triangle.vertices.contains(v) {
+            continue;
+        }
+
+        let vector_a_to_v = Position::sub(v, &triangle.vertices[0]);
+        let vector_b_to_v = Position::sub(v, &triangle.vertices[1]);
+        let vector_c_to_v = Position::sub(v, &triangle.vertices[2]);
+
+        // Cross product between A->B with A->V
+        let first_cross_product = get_cross_product_value(&vector_from_a_to_b, &vector_a_to_v);
+        // Cross product between B->C with B->V
+        let second_cross_product = get_cross_product_value(&vector_from_b_to_c, &vector_b_to_v);
+        // Cross product between C->A with C->V
+        let third_cross_product = get_cross_product_value(&vector_from_c_to_a, &vector_c_to_v);
+
+        result = result
+            || (first_cross_product < 0.0
+                && second_cross_product < 0.0
+                && third_cross_product < 0.0);
+    }
+
+    result
+}
+
+// Usually cross product results in another vector but since this is a two dimensional vector and
+// We just wanna use the value from this result we're going to return a numerical value
+fn get_cross_product_value(first_vector: &Position, second_vector: &Position) -> f32 {
+    first_vector.x * second_vector.y - first_vector.y * second_vector.x
+}
+
+// A convex polygon means that every inner angle doesn't have more than
+// 180 degrees
+fn is_polygon_convex(polygon: &Entity) -> bool {
+    let mut result = true;
+    for current_vertex_index in 0..polygon.vertices.len() {
+        let previous_vertex_index =
+            get_previous_cyclic_index(current_vertex_index, &polygon.vertices);
+        let next_vertex_index = get_next_cyclic_index(current_vertex_index, &polygon.vertices);
+
+        let previous_vertex = polygon.vertices[previous_vertex_index];
+        let current_vertex = polygon.vertices[current_vertex_index];
+        let next_vertex = polygon.vertices[next_vertex_index];
+
+        let first_vector = Position::sub(&previous_vertex, &current_vertex);
+        let second_vector = Position::sub(&next_vertex, &current_vertex);
+
+        result = result && get_cross_product_value(&first_vector, &second_vector) >= 0.0;
+    }
+
+    result
+}
+
+fn get_previous_cyclic_index(index: usize, vector: &[Position]) -> usize {
```

Comment:
What's this? What's a cyclic index?

Reply:
Answered via Slack. It's just we're getting previous vertex_index from a vector and current vertex index. In case the current is the first one, we get the last (works like a cyclic list).

Since the code doesn't ever talk about cyclic, and it's not explained, I think we should change the name of the function to something like "get_previous_vertex_index" or "get_previous_vertex".
I would use the 2nd with a comment above that we return the previous vertex by given vertex and vector.
The "cyclic" meaning just confuses, and the code is self-explanatory: there's an if that does that, it's easy to notice.

Reply:
I think i prefer the `get_previous_vertex_index` naming, since `get_previous_vertex` leads me to think that we return a vertex, nice suggestion! will make the change